files/external_node_v2.rb

```diff
@@ -196,8 +196,12 @@ def enc(certname)
   end
   res = http.start { |http| http.request(req) }
 
-  raise "Error retrieving node #{certname}: #{res.class}\nCheck Foreman's /var/log/foreman/production.log for more information." unless res.code == "200"
-  res.body
+  return res.body if res.code == '200'
+
+  raise Net::HTTPServerException if (500..599).include?(res.code.to_i)
```

Comment:
This fails with:

<pre>
/usr/share/ruby/net/http/exceptions.rb:5:in `initialize': wrong number of arguments (0 for 2) (ArgumentError)
        from /etc/puppetlabs/puppet/node.rb:201:in `exception'
        from /etc/puppetlabs/puppet/node.rb:201:in `raise'
        from /etc/puppetlabs/puppet/node.rb:201:in `enc'
        from /etc/puppetlabs/puppet/node.rb:348:in `block in <main>'
        from /usr/share/ruby/timeout.rb:66:in `timeout'
        from /etc/puppetlabs/puppet/node.rb:347:in `<main>'
</pre>

on ruby 2.0.0p598, it's also reproducible in irb.